cpp/locality/RawPoints.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2010-2019 The Regents of the University of Michigan
+// This file is from the freud project, released under the BSD 3-Clause License.
+
+#ifndef RAW_POINTS_H
+#define RAW_POINTS_H
+
+#include "NeighborQuery.h"
+#include "AABBQuery.h"
+
+/*! \file RawPoints.h
+    \brief Defines a simplest NeighborQuery object that actually farms out
+           querying logic to an AABBQuery.
+*/
+
+namespace freud { namespace locality {
+
+//! Dummy class to just contain minimal information and not actually query.
+/*! The purpose of this class is to support dynamic NeighborQuery object
+ *  resolution. Users may pass instances of this class instead of providing a
+ *  NeighborQuery to various compute functions throughout freud, which is an
+ *  indication that the function needs to compute its own NeighborQuery. That
+ *  logic, which is primary encapsulated in the NeighborComputeFunctional.h
+ *  file, helps provide a nice Python API as well.
+ */
+class RawPoints : public NeighborQuery
+{
+public:
+    RawPoints() {}
+
+    RawPoints(const box::Box& box, const vec3<float>* points, unsigned int n_points)
+        : NeighborQuery(box, points, n_points), aq(box, points, n_points) {}
```

Comment:
Should the AABBQuery be made on construction, or only if `querySingle` is called?

Maybe initialize `aq` as a null pointer. When `querySingle` is called, then construct the AABBQuery before querying if needed. Unless this needs to be parallel-safe... in which case just leave it as-is unless you have a better idea.